pkg/terra/db/db.go

```diff
@@ -0,0 +1,84 @@
+package db
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+type Chain struct {
+	ID        string
+	Nodes     []Node
+	Cfg       ChainCfg
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	Enabled   bool
+}
+
+type Node struct {
+	ID            int32
+	Name          string
+	TerraChainID  string
+	TendermintURL string `db:"tendermint_url"`
+	FCDURL        string `db:"fcd_url"`
+	CreatedAt     time.Time
+	UpdatedAt     time.Time
+}
+
+type ChainCfg struct {
+	BlocksUntilTxTimeout  null.Int
+	ConfirmMaxPolls       null.Int
+	ConfirmPollPeriod     *models.Duration
+	FallbackGasPriceULuna null.String
+	GasLimitMultiplier    null.Float
+	MaxMsgsPerBatch       null.Int
+}
+
+func (c *ChainCfg) Scan(value interface{}) error {
+	b, ok := value.([]byte)
+	if !ok {
+		return errors.New("type assertion to []byte failed")
+	}
+
+	return json.Unmarshal(b, c)
+}
+
+func (c ChainCfg) Value() (driver.Value, error) {
+	return json.Marshal(c)
+}
+
+// State represents the state of a given terra msg
+// Happy path: Unstarted->Broadcasted->Confirmed
+type State string
+
+var (
+	// Unstarted means queued but not processed.
+	// Valid next states: Broadcasted, Errored (sim fails)
+	Unstarted State = "unstarted"
+	// Broadcasted means included in the mempool of a node.
+	// Valid next states: Confirmed (found onchain), Errored (tx expired waiting for confirmation)
+	Broadcasted State = "broadcasted"
+	// Confirmed means we're able to retrieve the txhash of the tx which broadcasted the msg.
+	// Valid next states: none, terminal state
+	Confirmed State = "confirmed"
+	// Errored means the msg reverted in simulation OR the tx containing the message timed out waiting to be confirmed
+	// TODO: when we add gas bumping, we'll address that timeout case
+	// Valid next states, none, terminal state
+	Errored State = "errored"
+)
+
+type Msg struct {
+	ID         int64
+	ChainID    string `db:"terra_chain_id"`
+	ContractID string
+	State      State
+	Raw        []byte
```

Comment:
Renamed from `Msg` to avoid nesting, does `Raw` make sense?

Reply:
yep, maybe just comment that its a serialized msg